src/System.Management.Automation/FormatAndOutput/common/FormatViewGenerator_Complex.cs

```diff
@@ -681,23 +687,46 @@ private void DisplayLeaf(object val, List<FormatValue> formatValueList)
         /// </summary>
         /// <param name="val">Object to verify.</param>
         /// <param name="level">Current level of recursion.</param>
-        /// <returns></returns>
-        private static bool TreatAsLeafNode(object val, TraversalInfo level)
+        /// <param name="scalarTypesToExpand">Array of names of types that should not be rendered as scalars.</param>
+        /// <returns>True if it has to be treated as leaf node.</returns>
+        private static bool TreatAsLeafNode(object val, TraversalInfo level, string[] scalarTypesToExpand)
         {
             if (level.Level >= level.MaxDepth || val == null)
+            {
                 return true;
+            }
 
-            return TreatAsScalarType(PSObject.GetTypeNames(val));
+            return TreatAsScalarType(PSObject.GetTypeNames(val), scalarTypesToExpand);
         }
 
         /// <summary>
         /// Treat as scalar check.
         /// </summary>
         /// <param name="typeNames">Name of the type to check.</param>
+        /// <param name="scalarTypesToExpand">Array of names of types that should not be rendered as scalars.</param>
         /// <returns>True if it has to be treated as a scalar.</returns>
-        private static bool TreatAsScalarType(Collection<string> typeNames)
+        private static bool TreatAsScalarType(Collection<string> typeNames, string[] scalarTypesToExpand)
+        {
+            return DefaultScalarTypes.IsTypeInList(typeNames) || TypeIsScalarForComplexView(typeNames, scalarTypesToExpand);
```

Comment:
I'm not familiar with this part of the code, but it seems like adding `DateTime` and `DateTImeOffset` to `DefaultScalarTypes` would result in the desired behavior?